docs/contrib/contributing.rst

```diff
@@ -89,18 +89,32 @@ In order for a Pull Request to be considered, it has to meet these requirements:
 
 2.  Be accompanied by appropriate tests.
 
-3.  Be issued from a branch *other than master* (PRs coming from master will not
+3.  Be accompanied by appropriate documentation
+    :ref:`documentation <project-documentation>`.
+
+4.  Be issued from a branch *other than master* (PRs coming from master will not
     be accepted.)
 
 If not *all* of these requirements are met then the code should **not** be
 merged into the distribution, and need not even be reviewed.
 
+
+.. _project-documentation:
+
 Documentation
 -------------
 
-All code contributed to the user-facing standard library (the `nativelib/`
-directory) should come accompanied with documentation.
-Pull requests containing undocumented code will not be accepted.
+Pull requests, other than the most trivial, without documentation
+will not be accepted. Imagine yourself as the person merging trying
+to understand, use, merge, or maintain the contribution.
+
+All code containing changes
+potentially visible to end users or developers contributed to user-facing
```

Comment:
```suggestion
All code containing changes to user-facing
```
if they are user-facing, they are necessarily visible to end users or developers.

Reply:
Fixed & less stumbling-on-itself verbose.